### ISSUE TITLE
About Python2 and Windows 10

### ISSUE THREAD

**Issue by Suleman-Elahi** [smacke/subsync#4]
I tried it on my MPB with Python2 first and it failed. But it worked fine with Python3 an I used it on a whole movie file (more than 1 GB in size and 130 minutes long). Also, i tried it on Windows but couldn't install it. So, will it ever support Windows 10?

**Comment by smacke** [comment]
Hi there, what version of Python2 do you have, so I can try to reproduce? Also, if you have time, please copy+paste the stack trace as well.

Re. Windows 10 -- I'm not as familiar with Windows so it will take a bit of time (also lacking cycles currently), but eventually it will support Windows 10.

**Comment by postacik** [comment]
Windows 10 / Python 3

Installed ffmpeg with chocolaty.

When I run the following command it outputs the following and does nothing:

subsync 01.mkv -i 01.srt -o 01x.srt
extracting speech segments...
  0%|                                                                                     | 0/3575.168 [00:00<?, ?it/s]

I see that ffmpeg is running with the following options but it seems to be doing nothing.

ffmpeg.exe -i 01.mkv -f s16le -ac 1 -acodec pcm_s16le -ar 48000 -

**Comment by Suleman-Elahi** [comment]
@smacke Here is the CLI output in case of Python 2. Python version: 2.7.10

![screen shot 2019-02-26 at 8 14 50 pm](https://user-images.githubusercontent.com/24384996/53466023-33231580-3a76-11e9-8cef-ac12f3223e47.png)

**Comment by smacke** [comment]
@Suleman-Elahi looks like you have an older version of numpy. I made bumped the minimum version required to 1.12.0; you should get it automatically if you do
`pip install git+https://github.com/smacke/subsync`
again. Let me know how it works after trying that and hopefully I can close the issue.

@postacik I think you're probably seeing this error because `ffmpeg-python` uses *nix pipes for IPC, and if that's the case I think it won't work for Windows. I'll create a feature request; unfortunately it will take me some time to get to this. I've opened #8 to track this.

**Comment by steven-kraft** [comment]
Running with Python 3.7.0 on Windows 10 and ffmpeg installed with chocolaty works fine for me without Cygwin:

 ```subsync video.mp4 -i subs.srt -o sync.srt
extracting speech segments...
100%|██████████████████████████████████████████████████████████████████████████████████████████████▉| 288.2989583333333/288.299 [00:01<00:00, 226.11it/s]
...done.
computing alignments...
...done.
auditok (27634.000000000004, -642)
webrtcvad (19327.0, -641)
offset seconds: -6.420```

**Comment by postacik** [comment]
Python 3.6.6 on my side.

It does not work with .mkv files (?)

But after @steven-kraft 's message I tried with an mp4 and got the following successful result.
```
subsync x.mp4 -i 01.srt -o 02.srt
extracting speech segments...
100%|█████████████████████████████████████████████████████████████▉| 552.3795208333333/552.38 [00:02<00:00, 249.19it/s]
...done.
computing alignments...
...done.
auditok (4523.000000000001, -149324)
webrtcvad (3950.000000000002, -144591)
```

**Comment by postacik** [comment]
Additional info. 

I converted x.mp4 to mkv and it did not work.

There may be a problem with the ffmpeg options when the input file is mkv.

**Comment by steven-kraft** [comment]
@postacik Hmmm that's interesting. I converted my test video to mpv and it still worked for me.

**Comment by steven-kraft** [comment]
Additionally, I tried to install on Python 2.7.15 and it gave me some trouble when installing the webrtcvad dependency. However downloading and installing [Microsoft Visual C++ Compiler for Python 2.7](https://www.microsoft.com/en-us/download/details.aspx?id=44266) fixed the issue and it works fine for me:

```subsync video.mp4 -i subs.srt -o sync.srt
extracting speech segments...
100%|#########9| 288.298958333/288.299 [00:01<00:00, 190.98it/s]
...done.
computing alignments...
...done.
auditok (27634.000000000004, -642)
webrtcvad (19327.0, -641)
offset seconds: -6.420```

**Comment by smacke** [comment]
@steven-kraft I see, interesting... it looks like my hypothesis about `ffmpeg-python` not working on Windows is (at least partially) false. The plot thickens.

I think the py-webrtcvad dependency requires compiling some C code into a python extension, which could partially explain why the MS visual C++ python compiler is necessary. Not sure why it's only needed for Python 2.7 though; perhaps Windows comes with it for Python 3.x? I'm unfortunately not very familiar with how Windows does things.

Probably I will need to grab a Windows VM to thoroughly investigate all the different possibilities in order to fix #8.

**Comment by smacke** [comment]
I'm going to go ahead and close this, since I think 6f56d5c should fix @Suleman-Elahi's issue with not having `numpy.flip` available. Let's redirect Windows-related discussion to #8.
